test/extended/router/stress.go

```diff
@@ -270,34 +212,162 @@ var _ = g.Describe("[sig-network][Feature:Router][apigroup:route.openshift.io]",
-						break Wait
+						continue
+					}
+					if len(route.Status.Ingress) > 1 {
+						other++
+					}
+					// Find UnservableInFutureVersions condition.
+					if cond := findIngressCondition(ingress, routev1.RouteUnservableInFutureVersions); cond != nil {
+						unservableCondition++
+						o.Expect(ingress.Host).NotTo(o.BeEmpty())
+						o.Expect(ingress.Conditions).NotTo(o.BeEmpty())
+						o.Expect(cond.LastTransitionTime).NotTo(o.BeNil())
+						o.Expect(cond.Status).To(o.Equal(corev1.ConditionTrue))
 					}
 				}
-				o.Expect(writes).To(o.BeNumerically("<", 5))
-			}()
+				// if other routers are writing status, wait until we get a complete
+				// set since we don't have a way to tell other routers to ignore us
+				if unservableCondition < 3 && other%20 != 0 {
+					return false, nil
+				}
+				outputIngress(routes.Items...)
+				rv = routes.ResourceVersion
+				return true, nil
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("deploying a scaled out namespace scoped router that removes the UnservableInFutureVersions condition")
+			rsRemove, err := oc.KubeClient().AppsV1().ReplicaSets(ns).Create(
+				context.Background(),
+				scaledRouter(
+					"router-remove-condition",
+					routerImage,
+					[]string{
+						"-v=5",
+						fmt.Sprintf("--namespace=%s", ns),
+						// the contention tracker is resync / 10, so this will give us 2 minutes of contention tracking
+						"--resync-interval=20m",
+						"--name=conflicting",
+						"--debug-upgrade-validation-force-remove-condition",
+					},
+				),
+				metav1.CreateOptions{},
+			)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = waitForReadyReplicaSet(oc.KubeClient(), ns, rsRemove.Name)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("verifying that we stop writing conflicts rapidly")
+			verifyConflictingWrites(client, rv, 15*time.Second, 50, "")
+
+			g.By("clearing a single route's status")
+			route, err := client.Patch(context.Background(), "9", types.MergePatchType, []byte(`{"status":{"ingress":[]}}`), metav1.PatchOptions{}, "status")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("verifying that only get a few updates")
+			verifyConflictingWrites(client, route.ResourceVersion, 10*time.Second, 5, "9")
 		})
 	})
 })
 
+// verifyConflictingWrites verifies that the router stops writing an object in a specified amount of time.
+func verifyConflictingWrites(client v1.RouteInterface, rv string, observeTime time.Duration, writeLimit int, routeNameMatch string) {
+	writes := 0
+	w, err := client.Watch(context.Background(), metav1.ListOptions{Watch: true, ResourceVersion: rv})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	func() {
+		defer w.Stop()
+		timer := time.NewTimer(observeTime)
+		ch := w.ResultChan()
+	Wait:
+		for i := 0; ; i++ {
+			select {
+			case obj, ok := <-ch:
+				o.Expect(ok).To(o.BeTrue())
+				if routeNameMatch != "" {
+					if r, ok := obj.Object.(*routev1.Route); ok {
+						if r == nil || routeNameMatch != routeNameMatch {
```

Comment:
We're comparing the same variable. Typo? Should this be `r.Name !=`? (Later edit: looks like this code was always encoded this way.)